src/main/kotlin/org/danilopianini/gradle/mavencentral/Configuration.kt

```diff
@@ -75,7 +75,7 @@ fun Project.configureRepository(repoToConfigure: Repository) {
         publishing.repositories { repository ->
             repository.maven { mavenArtifactRepository ->
                 mavenArtifactRepository.name = repoToConfigure.name
-                mavenArtifactRepository.url = URI(repoToConfigure.url)
+                mavenArtifactRepository.url = repoToConfigure.url.map { URI(it) }.get()
```

Comment:
Why this change? It looks like you are breaking the laziness...

Reply:
I see no real alternative here: we're assigning a property to an ordinary mutable variable, so we need to access the final actual of the property.

Btw my edit was about merging classes `MavenRepositoryDescriptor` and `Repository` which were essentially the same thing, yet having no common super type. Here, the previous code was using the `url` variable of the former class, which was not supporting laziness. I replaced this variable by an homonymous property from the latter class, yet the logic of the code is unchanged: so, the property will certainly have a value when reaching this point in the code.

However, if you have a better way to write this, please enlighten me. Is the a way to add a callback to a property being assigned?